django/contrib/gis/gdal/raster/source.py

```diff
@@ -425,6 +425,25 @@ def warp(self, ds_input, resampling='NearestNeighbour', max_error=0.0):
 
         return target
 
+    def clone(self):
+        """Return a clone of this GDALRaster."""
+        if self.driver.name != 'MEM':
+            clone_name = os.path.join(VSI_FILESYSTEM_BASE_PATH, str(uuid.uuid4()))
```

Comment:
A clone of a file based raster, should probably be file based as well.  File based rasters can be very large and may frequently not fit into memory. So here we should probably construct a name on the filesystem, maybe in the same location. For the `warp` function, something along those lines is used if no explicit path is provided, see https://github.com/django/django/blob/a758017f13b84806eff4d8175bcbcff9302a8e69/django/contrib/gis/gdal/raster/source.py#L401.

The `VSI_FILESYSTEM_BASE_PATH` based name would result in a raster under the `/vsimem/` basepath on the filesystem.